client/checkout/blocks/fields.js

```diff
@@ -0,0 +1,97 @@
+/**
+ * External dependencies
+ */
+import {
+	Elements,
+	ElementsConsumer,
+	CardElement,
+} from '@stripe/react-stripe-js';
+import {
+	useEffect,
+	useState,
+} from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import generatePaymentMethod from './generate-payment-method.js';
+import confirmCardPayment from './confirm-card-payment.js';
+
+const WCPayFields = ( {
+	api,
+	stripe,
+	elements,
+	billing: {
+		billingData,
+	},
+	eventRegistration: {
+		onPaymentProcessing,
+		onCheckoutAfterProcessingWithSuccess,
+	},
+	emitResponse,
+} ) => {
+	const [ errorMessage, setState ] = useState( null );
+
+	// When it's time to process the payment, generate a Stripe payment method object.
+	useEffect(
+		() => onPaymentProcessing( () => {
+			if ( errorMessage ) {
+				return {
+					type: 'error',
+					message: errorMessage,
+				};
+			}
+
+			const cardElement = elements.getElement( CardElement );
+			const paymentElements = {
+				card: cardElement,
+			};
+
+			return generatePaymentMethod( api, paymentElements, billingData );
+		} ),
+	[ elements, stripe ] );
+
+	// Once the server has completed payment processing, confirm the intent of necessary.
+	useEffect(
+		() => onCheckoutAfterProcessingWithSuccess(
+			( { processingResponse: { paymentDetails } } ) => (
+				confirmCardPayment( api, paymentDetails, emitResponse )
+			)
+		),
+	[ elements, stripe ] );
+
+	// Checks whether there are errors within a field, and saves them for later reporting.
+	const checkForErrors = ( { error } ) => {
+		setState( error ? error.message : null );
+	};
+
+	const elementOptions = {
+		hidePostalCode: true,
+		classes: {
+			base: 'wcpay-card-mounted',
+		},
+	};
+
+	return <CardElement
+		options={ elementOptions }
+		onChange={ checkForErrors }
+	/>;
+};
+
+/**
+ * Wraps WCPayFields within the necessary Stripe consumer components.
+ *
+ * @param {Object} props All props given by WooCommerce Blocks.
+ * @returns {Object}     The wrapped React element.
+ */
+const ConsumableWCPayFields = ( { api, ...props } ) => (
+	<Elements stripe={ api.getStripe() }>
```

Comment:
As suggested by @nerrad, you'll likely want to check `props.activePaymentMethod` here to ensure that Stripe/WC Pay is not invoked when shopper uses a different gateway. 

I tested it with the following change:

```js
const ConsumableWCPayFields = ( { api, ...props } ) => {
	const { activePaymentMethod } = props;
	return ( activePaymentMethod === PAYMENT_METHOD_NAME ) && (
		<Elements stripe={ api.getStripe() }>
			<ElementsConsumer>
				{ ( { elements, stripe } ) => (
					<WCPayFields api={ api } elements={ elements } stripe={ stripe } { ...props } />
				) }
			</ElementsConsumer>
		</Elements>
	);
}
```

This fixes the issue I saw in testing - if WC Pay is activated as a gateway, and shopper selects a different gateway (e.g. `Check`), the Stripe elements form validation fires, showing a spurious "Please enter a valid credit card number" error :)

Reply:
Actually, my mistake was that I was trying to use `activePaymentMethod` in the effect, without indicating the dependency on it. https://github.com/Automattic/woocommerce-payments/pull/723/commits/9204ccd9fe84e1eb350799485d5c327f5e32e059 fixes this.

Your method would work when switching from Cheque to WCPay, but not when swiching back to Cheque, because the component would be already initialized.